src/main/java/ca/lambtoncollege/fsdm/s24/auction/service/AuctionService.java

```diff
@@ -82,15 +88,31 @@ public static Auction createAuction(String title, String description, String min
         auction.setState(Auction.State.Open);
         auction.setCreatedBy(user);
 
+        if (imagePart != null && imagePart.getSize() > 0) {
+            if (imagePart.getSize() > maxFileSize) {
+                errors.add("Image file size should be less than 1MB");
+            }
+            if (errors.isEmpty()) {
+                InputStream imageStream = imagePart.getInputStream();
+                String base64Image = Base64.getEncoder().encodeToString(imageStream.readAllBytes());
+                auction.setImage(base64Image);
+            }
+        }
+        if (!errors.isEmpty()) {
+            throw new ValidationException(errors);
+        }
+
         AuctionRepository.addAuction(auction);
 
         return auction;
     }
 
     public static Auction getAuction(int id) throws Exception {
-        var auction = AuctionRepository.getAuctionById(id);
-
-        if (auction == null) {
+        Auction auction = repository.getAuctionById(id);
+        if (auction != null && auction.getAuctionImage() != null) {
+            String base64Image = Base64.getEncoder().encodeToString(auction.getAuctionImage());
+            auction.setImageBase64(base64Image);
```

Comment:
See if you can move this code to AuctionRepository since all other `setX()` calls are being made there anyway